src/Middlewares/RequestResponseLoggingMidleware.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.IO;
+
+namespace fiskaltrust.Launcher.Middlewares
+{
+    public class RequestResponseLoggingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly RecyclableMemoryStreamManager _recyclableMemoryStreamManager;
+        public RequestResponseLoggingMiddleware(RequestDelegate next, ILoggerFactory loggerFactory)
+        {
+            _next = next;
+            _logger = loggerFactory.CreateLogger<RequestResponseLoggingMiddleware>();
+            _recyclableMemoryStreamManager = new RecyclableMemoryStreamManager();
+        }
+        public async Task Invoke(HttpContext context)
+        {
+            await LogRequest(context);
+            await LogResponse(context);
+        }
+
+        private async Task LogRequest(HttpContext context)
+        {
+            context.Request.EnableBuffering();
+
+            await using var requestStream = _recyclableMemoryStreamManager.GetStream();
+            await context.Request.Body.CopyToAsync(requestStream);
+
+            _logger.LogDebug("Http Request Information: Schema \"{Scheme}\" Host \"{Host}\" Path \"{Path}\" QueryString \"{QueryString}\" Body {RequestBodyStream}",
```

Comment:
We could check for the debug level before we log here, to reduce the number of string allocations (which can be quite expensive for large content sizes). Currently, we always iterate over the request/response and allocate the string, but we could do something like:
```
if(loglevel = debug) {
  // do all this stuff
}
```